TickSpec/ScenarioRun.fs

```diff
@@ -135,14 +135,16 @@ let invokeStep
     let tail =
         match bullets,table,doc with
         | Some xs,None,None ->
-            let p = ps.[ps.Length-1]
-            let t = p.ParameterType.GetElementType()
+            let p = if ps.Length > args.Length then ps.[args.Length].ParameterType
+                    else failwithf "Expected an array argument at position %d" args.Length
```

Comment:
This exception is misleading right? You are not really checking whether it is an array or not right now.